pre_commit/languages/dotnet.py

```diff
@@ -0,0 +1,108 @@
+import contextlib
+import glob
+import os.path
+import shutil
+from typing import Generator
+from typing import Sequence
+from typing import Tuple
+
+import pre_commit.constants as C
+from pre_commit.envcontext import envcontext
+from pre_commit.envcontext import PatchesT
+from pre_commit.envcontext import Var
+from pre_commit.hook import Hook
+from pre_commit.languages import helpers
+from pre_commit.prefix import Prefix
+from pre_commit.util import clean_path_on_failure
+from pre_commit.util import rmtree
+
+ENVIRONMENT_DIR = 'dotnetenv'
+BIN_DIR = 'bin'
+BUILD_CONFIG = 'Release'
+
+get_default_version = helpers.basic_get_default_version
+healthy = helpers.basic_healthy
+
+
+def get_env_patch(venv: str) -> PatchesT:
+
+    # https://github.com/dotnet/sdk/issues/9304
+    dotnet = shutil.which('dotnet')
+    assert dotnet, 'No dotnet executable found - please install one'
+    dotnet_root = os.path.dirname(os.path.realpath(dotnet))
+
+    return (
+        ('PATH', (os.path.join(venv, BIN_DIR), os.pathsep, Var('PATH'))),
+        ('DOTNET_ROOT', dotnet_root),
+    )
+
+
+@contextlib.contextmanager
+def in_env(prefix: Prefix) -> Generator[None, None, None]:
+    directory = helpers.environment_dir(ENVIRONMENT_DIR, C.DEFAULT)
+    envdir = prefix.path(directory)
+    with envcontext(get_env_patch(envdir)):
+        yield
+
+
+def install_environment(
+        prefix: Prefix,
+        version: str,
+        additional_dependencies: Sequence[str],
+) -> None:
+
+    helpers.assert_version_default('dotnet', version)
+    helpers.assert_no_additional_deps('dotnet', additional_dependencies)
+
+    envdir = helpers.environment_dir(ENVIRONMENT_DIR, version)
+    repodir = prefix.path(envdir)
+
+    with clean_path_on_failure(repodir):
+
+        build_dir = 'pre-commit-build'
+
+        # Build & pack nupkg file
+        helpers.run_setup_cmd(
+            prefix,
+            (
+                'dotnet', 'pack',
+                '--configuration', BUILD_CONFIG,
+                '--output', build_dir,
+            ),
+        )
+
+        # Determine the tool name
+        build_outputs = os.listdir(os.path.join(prefix.prefix_dir, build_dir))
+        assert (
+            len(build_outputs) == 1
+        ), f"Can't handle multiple build outputs. Got {build_outputs}"
```

Comment:
let's change this into an if statement with a raise, that'll make this always work even if they (for some reason) set `-0`